airflow/utils/file.py

```diff
@@ -108,46 +111,167 @@ def open_maybe_zipped(fileloc, mode='r'):
-            rel_file_path = os.path.join(os.path.relpath(root, str(base_dir_path)), str(file))
-            if any(p.search(rel_file_path) for p in patterns):
+            if match_pattern_fn(root, file, patterns):
                 continue
             yield str(abs_file_path)
 
 
+def _regex_syntax_compile_fn(
+    pattern: str, base_path: str, ignore_file_path: str
+) -> Optional[Union[Pattern, IgnoreRule]]:
+    """
+    Compile a regular expression from the supplied pattern and log a useful warning if it is invalid
+    :param pattern: the string pattern
+    :param base_path: the base directory where the pattern is defined, unused
+    :param ignore_file_path: the ignore file where the pattern was defined, used for logging warnings
+
+    :return: a regular expression Pattern
+    """
+    p: Optional[Union[Pattern, IgnoreRule]] = None
+    try:
+        p = re.compile(pattern)
+    except re.error as e:
+        log.warning(f"Ignoring invalid regex '{pattern}' from {ignore_file_path}: {e}")
+    return p
+
+
+def _get_regex_syntax_match_fn(
+    base_dir_path: str,
+) -> Callable[[str, str, List[Pattern]], bool]:
+    """
+    Return a function which can take a root, a candidate subpath and a list of regular expression
+    patterns for evaluation.
+    :param base_dir_path: the base path for DAG searching, supplied to the underlying function via closure
+
+    :return: a callable function which can be used to evaluate supplied ignore expressions
+    """
+
+    def match_function(root: str, subpath: str, patterns: List[Pattern]) -> bool:
+        test_path: str = os.path.join(os.path.relpath(root, str(base_dir_path)), subpath)
+        return any(p.search(test_path) for p in patterns)
+
+    return match_function
+
+
+def _glob_syntax_compile_fn(
+    pattern: str, base_path: str, ignore_file_path: str
+) -> Optional[Union[Pattern, IgnoreRule]]:
+    """Build an ignore rule from the supplied pattern and log a useful warning if it is invalid"""
+    rule = rule_from_pattern(pattern, base_path=Path(base_path).resolve())
+    if not rule:
+        # There are four reasons why a None rule might be returned by "rule_from_pattern":
+        # 1. The pattern string is empty or is a comment - we can ignore
+        # 2. The pattern contains three or more consecutive '*' chars - should warn
+        # 3. The pattern contains '**' not at beginning or end and it is not between '/' chars - should warn
+        # 4. The pattern is '/' which matches nothing - should warn
+        reason = ""
+        if '/' == pattern.strip():
+            reason = "will not match any files or directories"
+        elif '***' in pattern:
+            reason = "more than two consecutive '*'"
+        elif '**' in pattern and '/**/' not in pattern:
+            reason = "'**' must be between '/' chars when not at beginning or end of pattern"
+        if reason:
+            log.warning(f"Ignoring glob '{pattern}' from {ignore_file_path}: {reason}")
```

Comment:
```suggestion
            log.warning(f"Ignoring glob '%s' from %s: %s:', pattern, ignore_file_path,  reason)
```
Please avoid stirring formatting before passing to logger.